backend/src/app.ts

```diff
@@ -114,6 +114,36 @@ app.get('/buildings/:landlordId', async (req, res) => {
   }
 });
 
+app.get('/buildings/all/:landlordId', async (req, res) => {
+  const { landlordId } = req.params;
+  const buildingDocs = (await buildingsCollection.where('landlordId', '==', landlordId).get()).docs;
+  const buildings: ApartmentWithId[] = buildingDocs.map(
+    (doc) => ({ id: doc.id, ...doc.data() } as ApartmentWithId)
+  );
+
+  const pageData = await Promise.all(
```

Comment:
This entire code block for `pageData` is repeated in an existing route `app.get('/page-data/:page', async (req, res) => {...})`. Can you put it in a separate helper function and just call the function in both routes?